src/app/control.rs

```diff
@@ -369,87 +369,107 @@ pub mod client {
+        <C::Value as connect::Connect>::Future: Send + 'static,
+        <C::Value as connect::Connect>::Error: ::std::error::Error + Send + Sync + 'static,
+        B: Payload,
     {
         Layer { _p: PhantomData }
     }
 
-    impl<C> Clone for Layer<C>
-    where
-        C: svc::Stack<connect::Target> + Clone,
-        C::Value: connect::Connect,
-        <C::Value as connect::Connect>::Connected: Send + 'static,
-    {
+    impl<C, B> Clone for Layer<C, B> {
         fn clone(&self) -> Self {
-            layer()
+            Layer { _p: PhantomData }
         }
     }
 
-    impl<C> svc::Layer<Target, connect::Target, C> for Layer<C>
+    impl<C, B> svc::Layer<Target, connect::Target, C> for Layer<C, B>
     where
         C: svc::Stack<connect::Target> + Clone,
-        C::Value: connect::Connect,
+        C::Value: connect::Connect + Clone + Send + Sync  + 'static,
         <C::Value as connect::Connect>::Connected: Send + 'static,
+        <C::Value as connect::Connect>::Future: Send + 'static,
+        <C::Value as connect::Connect>::Error: ::std::error::Error + Send + Sync + 'static,
+        B: Payload,
     {
-        type Value = <Stack<C> as svc::Stack<Target>>::Value;
-        type Error = <Stack<C> as svc::Stack<Target>>::Error;
-        type Stack = Stack<C>;
+        type Value = <Stack<C, B> as svc::Stack<Target>>::Value;
+        type Error = <Stack<C, B> as svc::Stack<Target>>::Error;
+        type Stack = Stack<C, B>;
 
         fn bind(&self, connect: C) -> Self::Stack {
-            Stack { connect }
+            Stack {
+                connect,
+                _p: PhantomData,
+            }
         }
     }
 
     // === impl Stack ===
 
-    impl<C> svc::Stack<Target> for Stack<C>
+    impl<C, B> Clone for Stack<C, B>
+    where
+        C: Clone,
+    {
+        fn clone(&self) -> Self {
+            Stack {
+                connect: self.connect.clone(),
+                _p: PhantomData,
+            }
+        }
+    }
+
+    impl<C, B> svc::Stack<Target> for Stack<C, B>
     where
         C: svc::Stack<connect::Target> + Clone,
-        C::Value: connect::Connect,
+        C::Value: connect::Connect  + Clone + Send + Sync + 'static,
         <C::Value as connect::Connect>::Connected: Send + 'static,
+        <C::Value as connect::Connect>::Future: Send + 'static,
+        <C::Value as connect::Connect>::Error: ::std::error::Error + Send + Sync + 'static,
+        B: Payload,
     {
-        type Value = client::Connect<
+        type Value = http::h2::Connect<
             C::Value,
-            ::logging::ClientExecutor<&'static str, Addr>,
-            BoxBody,
+            B,
         >;
         type Error = C::Error;
 
         fn make(&self, target: &Target) -> Result<Self::Value, Self::Error> {
             let c = self.connect.make(&target.connect)?;
-            let h2 = target.builder.clone();
+            let _h2 = target.builder.clone();
```

Comment:
should this line be removed? how can we configure window/stream concurrency?

Reply:
Ah yea, it should be removed. hyper can add builder options to configure HTTP2 things as we need them. We currently don't configure anything (besides disabling push, which hyper also does).